src/selectors.js

```diff
@@ -74,29 +94,84 @@ const _buildRecords = (ents, key, opts) => {
 
 const defaultArr = [];
 
-const buildRecords = (ents = {}, key, opts = {}) => {
-  // console.log(`All. key: ${key}`, ents, opts)
-  if (!_.isObject(opts)) opts = {};
-  const { model, defaultJoins } = getModel(key);
-  const joinKeys = utils.composeArray(opts.joins, defaultJoins);
-  let final = enrichRecords(ents, key) || defaultArr;
-
-  joinKeys.forEach((k) => {
-    const mergeItems = enrichRecords(ents, k);
-    final = joins({
-      items: final,
-      mergeItems,
+const buildRecords = (entities = {}, key, opts = {}) => {
+  opts = _.isObject(opts) ? opts : {};
+  const { defaultJoins = [] } = getModel(key) || {};
+  let finalRecords = enrichRecords(entities, key) || defaultArr;
+  const joinsArray = utils.composeArray(defaultJoins, opts.joins);
+  const joinKeys = buildNestedStructure(joinsArray, key);
+
+  joinKeys.forEach(({ key: joinKey, children }) => {
+    const enrichedJoins = children.reduce((acc, child) => {
+      const parentRecords = acc.length
```

Comment:
kan je de parentRecords niet voor de reduce definieren? Da's toch eig altijd `key` (parent=key hier toch?)

Reply:
Als we op hetzelfde level al een andere join hebben gedaan is diei accumulator al gevuld met die eerdere join en gebruiken we die

Reply:
oh of je bedoelt het tweede deel van de ternary? DIe kan er idd buiten